### DISCUSSION TITLE
picking up configuration without rebuilding

### DISCUSSION THREAD

**Discussion by BharathKS77** [vercel/next.js#41214] [Help]
Hi ,

I am creating an executable from the next js project .I want the app to pick the config details without rebuilding the app .Can you please suggest ways to do it .

**Comment by icyJoseph** [comment]
If it is for environmental variables you can use `runtime` config, but that'll impact pre-rendering. https://nextjs.org/docs/api-reference/next.config.js/runtime-configuration

For other configuration options, then I think you might be out of luck, because answer is, it depends...

As far as I know, Next.js reads the `next.config.js` file, if available, at:

- Dev Server start (`next dev`)
- Build process start (`next build`)
- Production Server start (`next start`)

Now, some configurations apply during build time, some during dev mode, and others during runtime mode, for example:

- `reactStrictMode: true`, this only matters during dev time

Since this is not something I've done, this is as far as I can guide you. All I can give is an example:

If I build my app using `next/image` pointing to `https://picsum.photos`, then I'd have:

```js
/** @type {import('next').NextConfig} */
const nextConfig = {
  reactStrictMode: true,
  images: {
    domains: ["picsum.photos"],
  },
};

module.exports = nextConfig;
```

And when I build, everything would be fine, but then if I change the `domains` property or remove the config file altogether, and run `next start`, I won't see any images at all.

I think when building standalone mode, the `next.config.js` is copied so that you cannot modify it at all, see [here](https://nextjs.org/docs/advanced-features/output-file-tracing#automatically-copying-traced-files):

> Note: next.config.js is read during next build and serialized into the server.js output file. If the legacy [serverRuntimeConfig or publicRuntimeConfig options](https://nextjs.org/docs/api-reference/next.config.js/runtime-configuration) are being used, the values will be specific to values at build time.

Finally, one quirk which might be good to know, you can return a function from your `next.config.js` file. Though this doesn't work with `standalone` mode.

```js
/** @type {import('next').NextConfig} */
const nextConfig = {
  reactStrictMode: true,
  images: {
    domains: ["picsum.photos"],
  },
};

module.exports = (phase) => {
  console.log("We are at:", phase);
  return nextConfig;
};
```

You can read more about phases at the end of this page, https://nextjs.org/docs/api-reference/next.config.js/introduction

**Reply by BharathKS77** [reply]
Hi ,Sorry I dint get that .
